### PR TITLE
Problem: Out of date requirements

### PR DESCRIPTION
**Pull request by julianpistorius**
Solution: Freshen requirements. Also:
- 'Unpinned' versions from the `.in` files
- @c-mart's fix to `python-irodsclient` is now in the PyPI
- Removed `pip-tools` from requirements

Note: This contains a few major changes to dependencies which will
require testing & monitoring, notably:

keystoneauth1==2.21.0 -> 3.1.0
django==1.10.6 -> 1.11.3
django-cors-headers==0.12.0 -> 2.1.0
django-redis-cache==0.13.0 -> 1.7.1
pillow==2.5.3 -> 4.2.1
psycopg2==2.5.4 -> 2.7.3
pyjwt==1.4.0 -> 1.5.2
gevent==1.0.1 -> 1.2.2

P.S. This requires `pip-tools>=1.9.0` in Clank

## Checklist before merging Pull Requests
- ~[ ] New test(s) included to reproduce the bug/verify the feature~
- ~[ ] Documentation created/updated at [Example link to documentation](https://example.test/doc#new_section) to give context to the feature~
- [ ] Reviewed and approved by at least one other contributor.
- [ ] If necessary, include a snippet in CHANGELOG.md
- ~[ ] New variables supported in Clank~
- ~[ ] New variables committed to secrets repos~

### CODE REVIEW

**Comment by julianpistorius** [comment]
Will triage the failed tests. Don't merge yet.

File: features/instance.features/edit_instance.feature [hunk 1]
```diff
@@ -113,7 +113,7 @@ Feature: Launching & editing of an instance
     # TODO: Fix global_burn_rate
```
Comment (julianpistorius) [inline features/instance.features/edit_instance.feature]: Hahahahahaha....

**Comment by julianpistorius** [comment]
OK, I think we're ready @steve-gregory

**Comment by steve-gregory** [comment]
Thank you @julianpistorius

**Pull request merged**